static/js/browser2/statvar_group_node.tsx

```diff
@@ -183,47 +182,49 @@ export class StatVarHierarchyNodeHeader extends React.Component<
 interface ChildStatVarSectionPropType {
   data: StatVarNodeType[];
   pathToSelection: string[];
-  placeDcid: string;
-  placeName: string;
+  places: NamedPlace[];
   highlightedStatVar: React.RefObject<HTMLDivElement>;
 }
 
-export class ChildStatVarSection extends React.Component<
-  ChildStatVarSectionPropType
-> {
-  render(): JSX.Element {
-    return (
-      <div className="svg-node-child">
-        {this.props.data.map((statVar) => {
-          const isSelected =
-            this.props.pathToSelection.length === 1 &&
-            this.props.pathToSelection[0] === statVar.id;
-          return (
-            <div
-              key={statVar.id}
-              ref={isSelected ? this.props.highlightedStatVar : null}
-            >
+export function ChildStatVarSection(
+  props: ChildStatVarSectionPropType
+): JSX.Element {
+  const appContext = useContext(Context);
+  return (
+    <div className="svg-node-child">
+      {props.data.map((statVar) => {
+        const isSelected =
+          props.pathToSelection.length === 1 &&
+          props.pathToSelection[0] === statVar.id;
+        return (
+          <div
+            key={statVar.id}
+            ref={isSelected ? props.highlightedStatVar : null}
+          >
+            {appContext.StatVarHierarchyType ==
```

Comment:
i wonder if it would work as well to pass in the intended "child" component here using one of the [composition methods here](https://reactjs.org/docs/composition-vs-inheritance.html) instead of a global switch statement? you might have considered it already...

Reply:
Yeah, I did tried to use the composition method. The child component can only be dynamically generated, so the actual "prop" passed in would be the component definition. In that case, needs to force the components have same interface. For example, <StatVarCheckbox> should take places as well.. which does not quite make sense.

On the other hand, I feel the current way is not optimal too, as it's not fully generalized

Reply:
i see, agreed it doesn't make sense to pass in places in other contexts. this current method is ok too in the sense that possible variations are spelled out there.